pkgs/top-level/config.nix

```diff
@@ -25,11 +25,18 @@ let
       internal = true;
     };
 
+    # Should be replaced by importing <nixos/modules/misc/assertions.nix> in the future
+    # see also https://github.com/NixOS/nixpkgs/pull/207187
     warnings = mkOption {
       type = types.listOf types.str;
       default = [];
       internal = true;
     };
+    assertions = mkOption {
+      type = types.listOf types.unspecified;
```

Comment:
Don't use `types.unspecified`, also because it has poor merge semantics. `types.raw` or `types.anything` instead. Or better: Specify the full type

Reply:
This is simply copied out of the NixOS module and is only temporary, the full type should be specified in https://github.com/NixOS/nixpkgs/pull/207187#issuecomment-1631102876